src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderParser.cs

```diff
@@ -30,26 +31,40 @@ private sealed class ValueCreatorCache
             private static readonly FuncInstanceMethodInfo1<ValueCreatorCache, PrimitiveDataViewType, Func<RowSet, ColumnPipe>> _getCreatorVecCoreMethodInfo
                 = FuncInstanceMethodInfo1<ValueCreatorCache, PrimitiveDataViewType, Func<RowSet, ColumnPipe>>.Create(target => target.GetCreatorVecCore<int>);
 
-            private static volatile ValueCreatorCache _instance;
-            public static ValueCreatorCache Instance
+            private static volatile ValueCreatorCache _defaultInstance;
+            public static ValueCreatorCache DefaultInstance
             {
                 get
                 {
-                    return _instance ??
-                        Interlocked.CompareExchange(ref _instance, new ValueCreatorCache(), null) ??
-                        _instance;
+                    return _defaultInstance ??
+                        Interlocked.CompareExchange(ref _defaultInstance, new ValueCreatorCache(), null) ??
+                        _defaultInstance;
                 }
```

Comment:
suggestion: If InterlockedCompareExchange fails, we are returning null. It would make it slightly more readable.

Reply:
By reading [the docs on Interlocked.CompareExchange](https://docs.microsoft.com/en-us/dotnet/api/system.threading.interlocked.compareexchange?view=netcore-3.1) I think that the above could be even written as `return _instance ?? Interlocked.CompareExchange(...)` (i.e. without the last `??`) since the method already returns the content of the variable to update, even if it didn't updated it.

In any case, I am hesitant to modify this, since it seems [everywhere in the codebase where we return a variable updated by Interlocked.CompareExchange](https://github.com/dotnet/machinelearning/search?p=2&q=Interlocked.CompareExchange&unscoped_q=Interlocked.CompareExchange) we follow exactly the same pattern that is above. Since my knowledge of multi-threaded programming is fairly low, I'll just assume that the pattern is correct for some reason, and it's better to leave it that way.

---
In reply to: [429557109](https://github.com/dotnet/machinelearning/pull/5154#discussion_r429557109) [](ancestors = 429557109)

Reply:
You are right, you can just change it to `return _instance ?? Interlocked.CompareExchange(...) `.  The pattern that is used throughout seems unnecessary. We can change it some other time. 

---
In reply to: [429585239](https://github.com/dotnet/machinelearning/pull/5154#discussion_r429585239) [](ancestors = 429585239,429557109)